src/core/geometry/qgsgeometry.cpp

```diff
@@ -1595,6 +1595,12 @@ double QgsGeometry::distance( const QgsGeometry &geom ) const
     return -1.0;
   }
 
+  // avoid calling geos for trivial point-to-point distance calculations
+  if ( QgsWkbTypes::flatType( d->geometry->wkbType() ) == QgsWkbTypes::Point && QgsWkbTypes::flatType( geom.constGet()->wkbType() ) == QgsWkbTypes::Point )
```

Comment:
Can `wkbType()` not directly be called on geom? Or is the idea to explicitly skip the nullptr check in `QgsGeometry::wkbType()`?

Reply:
Good point.